core/src/entrypoint/api.rs

```diff
@@ -82,12 +83,19 @@ pub async fn run(config: config::ApiConfig) -> Result<()> {
 
     let addr = config.host;
     tracing::info!("api server listening on {}", addr);
-    let server =
-        axum::Server::bind(&addr).serve(app.into_make_service_with_connect_info::<SocketAddr>());
+    let server = axum::serve(
+        TcpListener::bind(&addr).await.unwrap(),
+        app.into_make_service_with_connect_info::<SocketAddr>(),
+    )
+    .into_future();
```

Comment:
These `.into_future()` calls were needed for some reason to make `tokio::try_join!(server, metrics_server)?;` happy 🤷‍♂️

Didn't look too much into it, but the error message was a bit much, so I chose to just "fix" it... 😉